### DISCUSSION TITLE
Hybrid deployment

### DISCUSSION THREAD

**Discussion by FrancescoFaenzi** [matanolabs/matano#8] [Q&A] [answered]
What components may I deploy on different CSP and eventually on-premise?
Thank you
Best

**Comment by shaeqahmed** [comment, marked as answer]
We are currently focused on building out the Matano platform to be hosted on AWS because of its popularity and the wide availability of mature serverless primitives. If you are a multi-cloud organization, you can still ingest logs from providers such as GCP, Azure, etc. assuming the data transfer costs are manageable for your use case. We will be releasing managed log sources for popular Cloud/SaaS providers to make onboarding new log sources as easy as possible.

In the future, we want to make it possible to  customize your data lake destination (to any cloud/on-prem S3-compatible object storage provider) and even bring your own query engine (Snowflake, etc.) in addition to offering other cloud deployment models for Matano. 

Since Matano is designed for cloud-oriented security teams that are looking to analyze security data from petabyte-scale sources  for cheap and without the ops overhead of maintaining a SIEM, we do not have plans to offer a completely on-premise deployment model. Matano is to designed leverage the commodity of cloud compute wherever possible to achieve the goals of low-cost, high-scale, zero-ops security log analytics.

Feel free to elaborate on your use case, and what you are looking to achieve using Matano :)

**Reply by FrancescoFaenzi** [reply]
Hello. It completely makes sense. The opportunity you have in roadmap to customize data lake and query engine is definitely important IMO. Organizations that chose to invest in a data lake technology already could adopt the framework without losing the money. Other organizations could better comply with regulatory requirements: not every CSP is "compliant to everything" in every country and flexibility is needed (sometimes compliance still mandates on-premise!).
